taiga_halo2/src/transaction.rs

```diff
@@ -397,7 +398,17 @@ fn test_transaction_creation() {
     let spend_note_1 = {
         let vp_data = pallas::Base::zero();
         // TODO: add real application logic vps and encode them to vp_data_nonhashed later.
-        let vp_data_nonhashed = pallas::Base::zero();
+        let app_logic_vps_description = vec![
+            trivail_vp_description.clone(),
+            trivail_vp_description.clone(),
+        ];
+        // Encode the app_logic_vps_description into vp_data_nonhashed
+        // The encoding method is flexible and defined in the application vp.
+        // Use poseidon hash to encode the two logic vps here
+        let vp_data_nonhashed = poseidon_hash(
+            app_logic_vps_description[0].get_compressed(),
+            app_logic_vps_description[1].get_compressed(),
+        );
```

Comment:
I'm a little confused by this line. We put a _hash_ of the VK into a variable called `nonhashed`.

Reply:
Right, maybe we can have a better name?
The hash we used here is to encode the VPs(VKs). We can have other encoding methods.
The `nonhashed` means that the `vp_data_nonhashed` will not be hashed/encoded in value-base

Reply:
What about:
- `app_vk` (instead of `application_vp`)
- `app_data` (instead of `vp_data`)
- `user_data` (instead of `vp_data_nonhashed`)

Reply:
I agree with the first two renames.
I'm not sure if we need data, which is not about the user, will be encoded in `vp_data_nonhashed`.